### ISSUE TITLE
Break Computational Graph

### ISSUE THREAD

**Issue by apdullahyayik** [shenweichen/DeepCTR-Torch#208]
At BaseModel class, while computing metric for after each batch, below approach (at line 262) leads to constructing autograd edge on cpu :

```Python
 y.cpu().data.numpy(), y_pred.cpu().data.numpy().astype("float64")))
```

However, breaking computational graph before these like below will enable the code run faster:

```Python
y.detach().cpu().numpy(), y_pred.detach().cpu().numpy().astype("float64")))
```

Besides, .data is an old usage, currently it is not required.
